### DISCUSSION TITLE
How to add spaces between the function name and the parenthesis

### DISCUSSION THREAD

**Discussion by ppshobi** [PHP-CS-Fixer/PHP-CS-Fixer#6910] [Q&A] [answered]
```diff
-    public function __invoke (array $addedValues, array $args, Context $context, ResolveInfo $info) {
+    public function __invoke(array $addedValues, array $args, Context $context, ResolveInfo $info) {
         return self::determineAddedValues($addedValues);
     }
```

I have to keep the space between the function name and the opening parenthesis. I have the following rules in the config

```
    'single_space_after_construct' => true,
    'no_spaces_after_function_name' => false,
```

<details><summary>Full config rules</summary>
<p>

$rules = [
    // standard
    '@PHP81Migration'                                  => true,
    '@PHP80Migration:risky'                            => true,
    '@PSR1'                                            => true,
    '@PSR2' => true,
    '@PSR12'                                           => true,
    '@PSR12:risky'                                     => true,
    '@Symfony'                                         => true,
    '@PhpCsFixer'                                      => true,
    'blank_line_before_statement'                      => false,
    'class_attributes_separation'                      => true,
    'cast_spaces'                                      => ['space' => 'none'],
    'concat_space'                                     => ['spacing' => 'one'],
    'escape_implicit_backslashes'                      => ['double_quoted' => true, 'heredoc_syntax' => true, 'single_quoted' => true],
    'multiline_whitespace_before_semicolons'           => true,
    'no_extra_blank_lines'                             => true,
    'no_superfluous_phpdoc_tags'                       => ['allow_mixed' => true, 'remove_inheritdoc' => true],
    'single_space_after_construct' => true,
    'no_spaces_after_function_name' => false,
    'not_operator_with_space'                          => false,
    'nullable_type_declaration_for_default_null_value' => true,
    'ordered_class_elements'                           => false,
    'ordered_imports'                                  => true,
    'phpdoc_to_comment'                                => ['ignored_tags' => ['var']],
    'php_unit_internal_class'                          => false,
    'php_unit_test_class_requires_covers'              => false,
    'yoda_style'                                       => false,
    'return_assignment'                                => false,
    'trailing_comma_in_multiline'                      => ['elements' => ['arrays']],
    'braces'                                           => [
        'allow_single_line_closure' => true,
        'position_after_functions_and_oop_constructs' => 'same',
    ],
    // risky
    'function_to_constant'                   => true,
    'is_null'                                => true,
    //'date_time_immutable'                    => true,§
    'declare_strict_types'                   => false, //TODO://enable it in files one by one
    'dir_constant'                           => true,
    'modernize_types_casting'                => true,
    'no_alias_functions'                     => true,
    'no_homoglyph_names'                     => true,
    'non_printable_character'                => ['use_escape_sequences_in_strings' => true],
    'no_php4_constructor'                    => true,
    'phpdoc_no_alias_tag'                    => false,
    'php_unit_construct'                     => true,
    'php_unit_dedicate_assert'               => ['target' => 'newest'],
    'php_unit_expectation'                   => ['target' => 'newest'],
    'php_unit_method_casing'                 => true,
    'php_unit_mock'                          => ['target' => 'newest'],
    'php_unit_mock_short_will_return'        => true,
    'php_unit_namespaced'                    => ['target' => 'newest'],
    'php_unit_no_expectation_annotation'     => ['target' => 'newest'],
    'php_unit_set_up_tear_down_visibility'   => true,
    'php_unit_test_case_static_method_calls' => true,
    'pow_to_exponentiation'                  => true,
    'random_api_migration'                   => true,
    'use_arrow_functions'                    => false,
];
$config = new PhpCsFixer\Config();
return $config->setRules($rules)
    ->setRiskyAllowed(true)
    ->setFinder($finder);

</p>
</details> 

but the cs fixer removes the space between. Any pointers would be great.

**Comment by kubawerlos** [comment]
Have you tried to tun fixer with `-vvv` to see what rule is removing the space?

**Comment by ppshobi** [comment, marked as answer]
thanks for the tip, I got `function_declaration` rule that adjusts these spaces, i disabled it explicitly and everything works
